kafka-rest/src/test/java/io/confluent/kafkarest/v2/KafkaConsumerManagerTest.java

```diff
@@ -381,27 +385,47 @@ public void onCompletion(List<TopicPartitionOffset> offsets, Exception e) {
         consumerManager.deleteConsumer(groupName, consumer.cid());
     }
 
-    @Test
-    public void testBackoffMsControlsPollCalls() throws Exception {
-        bootstrapConsumer(consumer);
-        consumerManager.readRecords(groupName, consumer.cid(), BinaryKafkaConsumerState.class, -1, Long.MAX_VALUE,
-                new ConsumerReadCallback<ByteString, ByteString>() {
-                    @Override
-                    public void onCompletion(
-                        List<ConsumerRecord<ByteString, ByteString>> records, Exception e) {
-                        actualException = e;
-                        actualRecords = records;
-                        sawCallback = true;
-                    }
-                });
-
-        // backoff is 250
-        Thread.sleep(100);
-        // backoff should be in place right now. the read task should be delayed and re-ran until the max.bytes or timeout is hit
-        assertEquals(1, consumerManager.delayedReadTasks.size());
-        Thread.sleep(100);
-        assertEquals(1, consumerManager.delayedReadTasks.size());
+  @Test
+  public void testBackoffMsControlsPollCalls() throws Exception {
+    bootstrapConsumer(consumer);
+    ArrayList<Instant> pollTimestamps = new ArrayList<>();
+    consumer.schedulePollTask(
+        new Runnable() {
+          @Override
+          public void run() {
+            pollTimestamps.add(Instant.now());
+            consumer.schedulePollTask(this);
+          }
+        });
+    consumerManager.readRecords(
+        groupName,
+        consumer.cid(),
+        BinaryKafkaConsumerState.class,
+        -1,
+        Long.MAX_VALUE,
+        (records, e) -> sawCallback = true);
+
+    TestUtils.waitForCondition(() -> sawCallback, "Consumer task finished.");
+    assertEquals(
+        "Expected (2 initial polls + (1s timeout / 50ms backoff) =) 22 consumer polls.",
+        22,
+        pollTimestamps.size());
```

Comment:
Honest question: Do you think it makes sense to bump up the backoff? I have no idea whether it's OK to always expect such a precise time-slicing from the consumer.

Reply:
Done.